extensions/wikia/Flags/FlagsHelper.class.php

```diff
@@ -143,11 +143,25 @@ public function shouldDisplayFlags() {
 			/* Don't display flags on edit pages that are content namespaces */
 			&& !in_array(
 				$wgRequest->getVal( 'action', 'view' ),
-				[ 'edit', 'formedit' , 'history' ]
+				[ 'edit', 'formedit' , 'history', 'visualeditor' ]
 			);
 	}
 
 	/**
+	 * Checks if flags can be edited on current page to decide whether include edit modal
+	 * @return bool
+	 */
+	public function shouldDisplayFlags() {
```

Comment:
I would change this to "areFlagsEditable" and add one more condition:
* check for protection status using User::isAllowed('edit')